### PR TITLE
Upgrade schema

### PR DESCRIPTION
**Pull request by michamos**
## Description

This upgrades the schemas to version 51.*

## Checklist:

- [x] I have all the information that I need (if not, move to `RFC` and look for it).
- [x] I linked the related issue(s) in the corresponding commit logs.
- [x] I wrote [good commit log messages](https://github.com/torvalds/subsurface-for-dirk/blob/5f15ad5a86ada3c5e574041a5f9d85235322dabb/README#L92-L119).
- [x] My code follows the code style of this project.
- [x] I've added any new docs if API/utils methods were added.
- [ ] I have updated the existing documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by michamos** [comment]
Tests are still failing because too much needs to be changed on schema/dojson upgrades, and the diffs are huge so it's hard to find what has to be amended. I spent 1.5 days trying to modify the failing tests without complete success.

The problem is that they are testing too much IMHO: whether dojson produces the right output should be (and is extensively!) tested in dojson, not hepcrawl. Instead, hepcrawl should have tests asserting that dojson has been called correctly and that the output is reasonable (e.g. number of records produced is correct and they all contain, say, a `document_type` or some other required field).

@david-caro Something to discuss when you come back.

**Comment by jacquerie** [comment]
> The problem is that they are testing too much IMHO

:+1:

This style of testing adds strict coupling between different parts of the system, and moreover doesn't lead to better confidence in the quality of the software.

File: tests/functional/desy/fixtures/ftp_server/DESY/desy_collection_records.xml [hunk 1]
```diff
@@ -1,926 +1,363 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <collection xmlns="http://www.loc.gov/MARC21/slim">
-    <record>
-        <controlfield tag="001">111111</controlfield>
```
Comment (david-caro) [inline tests/functional/desy/fixtures/ftp_server/DESY/desy_collection_records.xml]: Why did you get rid of the control fields?

File: tests/functional/desy/fixtures/desy_ftp_records_expected.json [hunk 2]
```diff
@@ -1,2168 +1,466 @@
...
-    "inspire_categories": [
-        {
-            "source": "undefined",
-            "term": "Accelerators"
+[
+    {
+        "core": true, 
+        "documents": [
+            {
+                "url": "http://bib-pubdb1.desy.de/record/389977/files/desy-thesis-17-035.title.pdf",
```
Comment (david-caro) [inline tests/functional/desy/fixtures/desy_ftp_records_expected.json]: Is this how they will pass us the ffts? Will it always be an external url? (I was actually expecting to have a relative path in the ftp server...).

Comment (david-caro) [reply tests/functional/desy/fixtures/desy_ftp_records_expected.json]: Will this url be open to the world or will it need some authentication?

File: tests/functional/desy/fixtures/ftp_server/DESY/desy_collection_records.xml (hunk 1 above)
Comment (michamos) [reply tests/functional/desy/fixtures/ftp_server/DESY/desy_collection_records.xml]: I took real batchupload records. Those were inserts, so didn't have any controlfield yield.

File: tests/functional/desy/fixtures/desy_ftp_records_expected.json (hunk 2 above)
Comment (michamos) [reply tests/functional/desy/fixtures/desy_ftp_records_expected.json]: I have no idea how they will provide files. Isn't that something you discussed with them? Here the files are publicly available theses that were provided through HTTP, but you probably want to modify the test to take files from FTP.

File: tests/functional/desy/fixtures/ftp_server/DESY/desy_collection_records.xml (hunk 1 above)
Comment (david-caro) [reply tests/functional/desy/fixtures/ftp_server/DESY/desy_collection_records.xml]: okok, I'll work around that 👍

File: tests/functional/desy/fixtures/desy_ftp_records_expected.json (hunk 2 above)
Comment (david-caro) [reply tests/functional/desy/fixtures/desy_ftp_records_expected.json]: @fschwenn @ksachs Hey! can you clarify for me how will the fft files be passed to be parsed by hepcrawl? There's really no limitation, but I want to make sure that we properly support and test whatever methods you would like (passing an external url, a relative path inside the ftp, ...)

Thanks!

Comment (david-caro) [reply tests/functional/desy/fixtures/desy_ftp_records_expected.json]: I will go on with this pr using the url for now, will adapt it later.

Comment (ksachs) [reply tests/functional/desy/fixtures/desy_ftp_records_expected.json]: DESY internal stuff (PubDB) should always be public accessible URLs.

If we use the DESY spider for external stuff, as long as the 'real' spider for that publisher is not available, if depends on how the publisher provides the full-text. 
I guess we will have the real spider for those that push to our FTP server.
The rest is public URLs.

Comment (david-caro) [reply tests/functional/desy/fixtures/desy_ftp_records_expected.json]: @ksachs so, just to make sure, we will have:
* ftp relative paths (maybe for some of the publishers that push to DESY ftp)
* public urls  (for everybody else)

No?

Comment (ksachs) [reply tests/functional/desy/fixtures/desy_ftp_records_expected.json]: I think so.

Comment (david-caro) [reply tests/functional/desy/fixtures/desy_ftp_records_expected.json]: Perfect 👍

**Pull request merged**